### PR TITLE
PTNFLY-1790 adds checklist to PR template

### PR DESCRIPTION
**Pull request by andresgalante**
## Description

The goal is to make contributors aware that contribution should follow some standards, for example been responsive and cross browser compatible.

Thats why I've added a checklist to the PR template. This doesn't apply to all PRs, thats why I've marked it as optional. If the PR is relevant to a UI change, then it should follow the standards.

[Link to Jira issue](https://patternfly.atlassian.net/browse/PTNFLY-1790)
## Link to a preview

https://github.com/andresgalante/patternfly/blob/PTNFLY-1790-PR-template/.github/PULL_REQUEST_TEMPLATE.md

@dlabrecq @cardosogabriel @LHinson @bleathem can you please review this one?  👀

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -16,3 +16,17 @@ For example: Here is a [link to Alerts on rawgit](https://rawgit.com/patternfly/
 This is an image:
 
 ![Image description](http://placehold.it/350x150)
+
+## PR checklist (optional if relevant to the PR)
```
Comment (cardosogabriel) [inline .github/PULL_REQUEST_TEMPLATE.md]: Text is a bit confusing here. I'd say:
PR checklist (if relevant)

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -16,3 +16,17 @@ For example: Here is a [link to Alerts on rawgit](https://rawgit.com/patternfly/
...
+- [ ] **Cross browser**: works in IE10
+- [ ] **Cross browser**: works in IE11
+- [ ] **Cross browser**: works in Edge
+- [ ] **Cross browser**: works in Chrome
+- [ ] **Cross browser**: works in Firefox
+- [ ] **Cross browser**: works in Safari
+- [ ] **Cross browser**: works in Opera
+- [ ] **Responsive**: works in extra small, small, medium and large view ports.
+- [ ] **Code guidelines**: Follows [PatternFly Code Guide](http://codeguide.patternfly.org/)
+- [ ] **Preview the PR**: An imagine or an URL for designer to preview this PR is provided
```
Comment (cardosogabriel) [inline .github/PULL_REQUEST_TEMPLATE.md]: Typo here: An image

**Comment by cardosogabriel** [comment]
Checklist is super useful. I was not testing on Opera, for example. Thanks for this!

**Comment by andresgalante** [comment]
@cardosogabriel thanks!

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (jeff-phillips-18) [reply .github/PULL_REQUEST_TEMPLATE.md]: Along the same lines... "a URL"

Comment (dlabrecq) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm curious how folks go about testing Edge?

I have a virtual machine setup on my Mac to test IE10, but don't have the resources to test every Microsoft product/version. Virtual machines eat up a lot of precious space on my hard drive, so I don't know if installing multiple versions of Windows is realistic?

**Comment by dlabrecq** [comment]
Could we add a JIRA as the synopsis of the PR? For example, "PTNFLY-1698 What's New Wordpress Template Font Update”. This would provide me with more context when reviewing the PR. I could also see any comments made with the story.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -16,3 +16,17 @@ For example: Here is a [link to Alerts on rawgit](https://rawgit.com/patternfly/
 This is an image:
 
 ![Image description](http://placehold.it/350x150)
+
+## PR checklist (if relevant)
+
+- [ ] **Cross browser**: works in IE9
+- [ ] **Cross browser**: works in IE10
+- [ ] **Cross browser**: works in IE11
+- [ ] **Cross browser**: works in Edge
```
Comment (dlabrecq) [inline .github/PULL_REQUEST_TEMPLATE.md]: I'm curious how folks go about testing Edge?

I have a virtual machine setup on my Mac to test IE10, but don't have the resources to test every Microsoft product/version. Virtual machines eat up a lot of precious space on my hard drive, so I don't know if installing multiple versions of Windows is realistic?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 4]
```diff
@@ -1,7 +1,7 @@
 ## Description
 Write a few sentences describing the overall goals of the pull request's commits.
 
-(optional) Include related PRs or Issues
+(optional) Include related PRs, Issues and or Jira stories.
```
Comment (andresgalante) [inline .github/PULL_REQUEST_TEMPLATE.md:4]: @dlabrecq what do you think?

Comment (dlabrecq) [reply .github/PULL_REQUEST_TEMPLATE.md:4]: Great!

Can we also add a title for each section? I find myself adding "## Changes" when describing my changes, for example.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
 
-(optional) Include a list of changes:
+# Changes
```
Comment (andresgalante) [inline .github/PULL_REQUEST_TEMPLATE.md]: @dlabrecq is this what you had in mind?

**Comment by andresgalante** [comment]
@dlabrecq I don't know why it didn't get to the thread your concerns about edge and different IEs. You are right, not everyone has virtual machines or is lucky enough to have browserstack.

I think people should be aware of those requierements. If they can't provide it they can leave that item unchecked and a reviewer can check it for them. What do you think?

**Comment by dlabrecq** [comment]
Yes, I found it on browserstack. I had not been testing with it, but answered my question. Thanks.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 5 above)
Comment (dlabrecq) [reply .github/PULL_REQUEST_TEMPLATE.md]: Yes, thank you. I think there needs to be two '#' characters, tho -- this title is larger than others.

**Comment by dlabrecq** [comment]
LGTM

**Comment by andresgalante** [comment]
Can we merge this one?

**Comment by LHinson** [comment]
looping @jgiardino on this.

**Comment by jgiardino** [comment]
Looks good to me, too

**Comment by srambach** [comment]
I like it. Since there is a whole section for a Rawgit link/image, do you still want the last checkbox in addition to that? It's not bad and is an extra reminder, but is redundant.

**Comment by andresgalante** [comment]
@srambach you are right, that might work. Lets merge this and try it, if we see it fails we iterate.

Thanks a lot for the review! 😄

**Pull request merged**